### PR TITLE
Remove unused catch blocks.

### PR DESCRIPTION
**Pull request by NightlyNexus**
Writing to a Buffer does no I/O.

### CODE REVIEW

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java [hunk 1]
```diff
@@ -112,7 +101,7 @@ public void interceptAsync(@Nonnull final Operation operation, @Nonnull final Ap
     this.httpCall = null;
   }
 
-  private Call httpCall(Operation operation) throws IOException {
+  Call httpCall(Operation operation) {
```
Comment (NightlyNexus) [inline apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java]: unrelated visibility change obviates the need for a synthetic accessor method. will revert if unwanted.

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java [hunk 2]
```diff
@@ -134,16 +123,24 @@ private Call httpCall(Operation operation) throws IOException {
...
   }
 
-  public static String cacheKey(RequestBody requestBody) throws IOException {
+  public static String cacheKey(RequestBody requestBody) {
     Buffer hashBuffer = new Buffer();
-    requestBody.writeTo(hashBuffer);
+    try {
+      requestBody.writeTo(hashBuffer);
+    } catch (IOException e) {
+      throw new AssertionError(e);
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java]: why do we need IOException to be wrapped into Assert exception?

**Comment by sav007** [comment]
Sorry I don't understand why we convert IOException to AssertException. Could you please elaborate what are you trying to fix here?

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java (hunk 2 above)
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java]: Never mind I think I got what you mean

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java [hunk 3]
```diff
@@ -134,16 +123,24 @@ private Call httpCall(Operation operation) throws IOException {
...
   }
 
-  private RequestBody httpRequestBody(Operation operation) throws IOException {
+  private RequestBody httpRequestBody(Operation operation) {
     JsonAdapter<Operation> adapter = new OperationJsonAdapter(moshi);
     Buffer buffer = new Buffer();
-    adapter.toJson(buffer, operation);
+    try {
+      adapter.toJson(buffer, operation);
+    } catch (IOException e) {
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java]: Could we log in case of error, I know it should never happen but lets add logs here:
`logger.e(e, "Failed to prepare http call");`

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java [hunk 4]
```diff
@@ -134,16 +123,24 @@ private Call httpCall(Operation operation) throws IOException {
...
     return RequestBody.create(MEDIA_TYPE, buffer.readByteString());
   }
 
-  public static String cacheKey(RequestBody requestBody) throws IOException {
+  public static String cacheKey(RequestBody requestBody) {
     Buffer hashBuffer = new Buffer();
-    requestBody.writeTo(hashBuffer);
+    try {
+      requestBody.writeTo(hashBuffer);
+    } catch (IOException e) {
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java]: Could we log in case of error, I know it should never happen but lets add logs here:
`logger.e(e, "Failed to prepare http call");`

File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java (hunk 3 above)
Comment (NightlyNexus) [inline apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java]: I'd rather not add ambiguity to code by adding cases that can't happen.
If you'd like the change, I'll close this, and any other contributor is free to copy-paste this and modify it.

**Comment by NightlyNexus** [comment]
Feel free to reuse the code.

**Comment by sav007** [comment]
@NightlyNexus why this PR was closed?
